fmf/_compat_jsonschema.py

```diff
@@ -0,0 +1,51 @@
+"""Compatibility layer for jsonschema validation."""
+
+from typing import Any, Optional
+
+import jsonschema
+from jsonschema.validators import Draft4Validator
+
+
+def get_validator(
+        schema: Any,
+        schema_store: Optional[dict[str, Any]] = None
+        ) -> Draft4Validator:
+    """Create a validator instance based on available jsonschema version."""
+    # Validate schema is a dict
+    if not isinstance(schema, dict):
+        from fmf.utils import JsonSchemaError
+        raise JsonSchemaError(f'Invalid schema type: {type(schema)}. Schema must be a dictionary.')
```

Comment:
Probably this check can be removed when `mypy` is added. Should we add a todo note for that?